src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Authentication.cs

```diff
@@ -13,16 +13,113 @@
 using AspNet.Security.OpenIdConnect.Server;
 using JetBrains.Annotations;
 using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 
 namespace OpenIddict.Infrastructure {
     public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TScope, TToken> : OpenIdConnectServerProvider
         where TUser : class where TApplication : class where TAuthorization : class where TScope : class where TToken : class {
+        public override async Task ExtractAuthorizationRequest([NotNull] ExtractAuthorizationRequestContext context) {
+            var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
+
+            // Reject requests using the unsupported request parameter.
+            if (!string.IsNullOrEmpty(context.Request.GetParameter(OpenIdConnectConstants.Parameters.Request))) {
+                services.Logger.LogError("The authorization request was rejected because it contained " +
+                                         "an unsupported parameter: {Parameter}.", "request");
+
+                context.Reject(
+                    error: OpenIdConnectConstants.Errors.RequestNotSupported,
+                    description: "The request parameter is not supported.");
+
+                return;
+            }
+
+            // Reject requests using the unsupported request_uri parameter.
+            if (!string.IsNullOrEmpty(context.Request.RequestUri)) {
+                services.Logger.LogError("The authorization request was rejected because it contained " +
+                                         "an unsupported parameter: {Parameter}.", "request_uri");
+
+                context.Reject(
+                    error: OpenIdConnectConstants.Errors.RequestUriNotSupported,
+                    description: "The request_uri parameter is not supported.");
+
+                return;
+            }
+
+            // If a request_id parameter can be found in the authorization request,
+            // restore the complete authorization request stored in the user session.
+            if (!string.IsNullOrEmpty(context.Request.GetRequestId())) {
```

Comment:
I'm fine with this new dependency now, we should make it clear in the readme that we have to register the session middelware. If distributed cache will be requested in the future, we can add an abstraction here to allow customizing the request storage